src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -47,59 +68,78 @@ export class RuntimeLanguageContext {
             sh: 'shell',
             shellscript: 'shell',
             sql: 'sql',
+            shell: 'shell',
+            jsx: 'jsx',
+            tsx: 'tsx',
+            plaintext: 'plaintext',
         })
-        this.supportedLanguageExtensionMap = createConstantMap<CodeWhispererConstants.SupportedLanguage, string>({
+        this.supportedLanguageExtensionMap = createConstantMap<CodewhispererLanguage, string>({
             java: 'java',
             python: 'py',
-            javascriptreact: 'jsx',
+            jsx: 'jsx',
             javascript: 'js',
             typescript: 'ts',
-            typescriptreact: 'tsx',
+            tsx: 'tsx',
             csharp: 'cs',
             c: 'c',
-            c_cpp: 'cpp',
             cpp: 'cpp',
             go: 'go',
             kotlin: 'kt',
             php: 'php',
             ruby: 'rb',
             rust: 'rs',
             scala: 'scala',
-            sh: 'sh',
-            shellscript: 'sh',
+            shell: 'sh',
             sql: 'sql',
+            plaintext: 'txt',
         })
+    }
 
-        const values = Array.from(this.supportedLanguageMap.values())
-        const keys = Array.from(this.supportedLanguageMap.keys())
-        values.forEach(item => this.supportedLanguageSet.add(item))
-        keys.forEach(item => this.supportedLanguageSet.add(item))
+    /**
+     * Only used when invoking CodeWhisperer service API, for telemetry usage please use mapToCodewhispererLanguage
+     */
+    public mapToCodeWhispererRuntimeLanguage(language: CodewhispererLanguage): CodewhispererLanguage {
```

Comment:
why does it accept a `CodewhispererLanguage` and also return a `CodewhispererLanguage`? seems strange.

- this seems like it's doing a "normalize" operation, i.e. it's cleaning up values rather than mapping them
- this is in a CW module, in a CW class, so why repeat "codewhisperer" in the function name?

```suggestion
    public toRuntimeLanguage(language: CodewhispererLanguage): CodewhispererLanguage {
```